src/batVect.mli

```diff
@@ -621,6 +628,13 @@ val find : ('a -> bool) -> 'a t -> 'a
     @raise Not_found if there is no value that satisfies [p] in the
     vect [a]. *)
 
+val find_opt : ('a -> bool) -> 'a t -> 'a option
+(** [find p a] returns [Some x] if [x] is the first element of vect [a]
```

Comment:
I kept the old variable names here for consistency with the Make variable naming choice (I did not update it when I updated the variables around the normal `find` function; arguably this would be nicer, but la flemme.), but I fixed the other issues. thanks!